### PR TITLE
Downgrade Jansi library to prevent kcadm exception on Windows

### LINKED ISSUE #21851: v22.0.1 - Windows: kcadm.bat gives Java Exception
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/cli

### Describe the bug

I have installed v22.0.1 and when trying to run the kcadm.bat to import realms (and other items) the batch file will not run. Even running it by itself (to display the help) will give a Java Exception.

PS C:\Development\Test\manual_installs\keycloak-nightly\bin> .\kcadm.bat
Exception in thread "main" java.lang.NoClassDefFoundError: org/fusesource/jansi/WindowsAnsiOutputStream
        at org.jboss.aesh.console.settings.SettingsImpl.getTerminal(SettingsImpl.java:381)
        at org.jboss.aesh.console.Console.init(Console.java:190)
        at org.jboss.aesh.console.Console.<init>(Console.java:118)
        at org.jboss.aesh.console.AeshConsoleImpl.<init>(AeshConsoleImpl.java:98)
        at org.jboss.aesh.console.AeshConsoleBuilder.create(AeshConsoleBuilder.java:160)
        at org.keycloak.client.admin.cli.KcAdmMain.main(KcAdmMain.java:74)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 6 more

Java Version: 
openjdk 17.0.8 2023-07-18 LTS
OpenJDK Runtime Environment Zulu17.44+15-CA (build 17.0.8+7-LTS)
OpenJDK 64-Bit Server VM Zulu17.44+15-CA (build 17.0.8+7-LTS, mixed mode, sharing)

Windows Version:
Edition	Windows 10 Enterprise
Version	22H2
Installed on	‎6/‎9/‎2022
OS build	19045.3086
Experience	Windows Feature Experience Pack 1000.19041.1000.0

Running on the same machine with the same version of Java using Keycloak 21.1.2 does not show this error.

The server seems to be running fine with this version of Java. I can access the server, login, configure, etc.

### Version

22.0.1

### Expected behavior

The admin cli will run as expected.

### Actual behavior

PS C:\Development\Test\manual_installs\keycloak-nightly\bin> .\kcadm.bat
Exception in thread "main" java.lang.NoClassDefFoundError: org/fusesource/jansi/WindowsAnsiOutputStream
        at org.jboss.aesh.console.settings.SettingsImpl.getTerminal(SettingsImpl.java:381)
        at org.jboss.aesh.console.Console.init(Console.java:190)
        at org.jboss.aesh.console.Console.<init>(Console.java:118)
        at org.jboss.aesh.console.AeshConsoleImpl.<init>(AeshConsoleImpl.java:98)
        at org.jboss.aesh.console.AeshConsoleBuilder.create(AeshConsoleBuilder.java:160)
        at org.keycloak.client.admin.cli.KcAdmMain.main(KcAdmMain.java:74)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 6 more

### How to Reproduce?

Install v22.0.1 on Windows and go to the bin folder, run "kcadm.bat". 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Pepo48**
Closes #21851

The Quarkus Platform BOM that we introduced in #21426 contains `jansi` version 2.4.0. `WindowsAnsiOutputStream`[ isn't though present in the later versions](https://github.com/fusesource/jansi/blob/5f5f95c78252ea661d8be55930e3a7c18da9bea8/changelog.md?plain=1#L208) of the library.  `org.jboss.aesh` seems to use the older `WindowsAnsiOutputStream`, which is replaced by `WindowsAnsiPrintStream` in the more recent versions. Therefore, the easiest way to overcome the exception is to override back to the compatible version 1.18.

### CODE REVIEW

**Review by vmuzikar — COMMENTED** [review]
@Pepo48 Thanks for the fix!

I'm not sure if downgrading is the long-term solution for us. The missing `WindowsAnsiOutputStream` is not considered a bug from Jansi side and migrating to `WindowsAnsiPrintStream` is probably not something we can easily do. Sticking to old version doesn't seem like an option even from the potential CVEs perspective. Maybe we could use some [alternative solution](https://github.com/micronaut-projects/micronaut-core/issues/1521#issuecomment-483297610) for now?

**Comment by Pepo48** [comment]
> Maybe we could use some [alternative solution](https://github.com/micronaut-projects/micronaut-core/issues/1521#issuecomment-483297610) for now?

Discussed offline. Since this is not a downgrade per se, but rather a revert to the state before introducing the Quarkus Platform BOM, we can live with it for now.

In the future we plan to get rid of the ˇorg.jboss.aesh` dependency, which is actually the core of the issue here.

**Review by vmuzikar — APPROVED** [review]
LGTM, thank you @Pepo48!

I'd like to wait for an additional review from someone with access to a Win machine. Maybe @ssilvert?

**Review by mabartos — APPROVED** [review]
Unfortunately, I wasn't able to reproduce the issue with the latest upstream main, but as it's pretty difficult to reproduce it and the others have encountered the problem, the issue is valid for sure. 

I tried this approach on the Win machine, and everything seems to be working as expected. I executed the command multiple times, made some experiments, and I've never faced the exception.

@Pepo48 @vmuzikar Thus, LGTM!

**Comment by Pepo48** [comment]
no worries @mabartos, I couldn't reproduce it on an OpenStack VM as well - https://github.com/keycloak/keycloak/issues/21851#issuecomment-1644166143. I've got "luck" only with my personal W11 machine.

**Review by ahus1 — COMMENTED** [review]
Better late than never: I tested 22.0.1 and reproduced the problem on a Windows machine, and also reproduced that this PR fixes it. Thanks!

**Comment by Pepo48** [comment]
> Better late than never: I tested 22.0.1 and reproduced the problem on a Windows machine, and also reproduced that this PR fixes it. Thanks!

Thanks for the review, @ahus1 🙂

**Pull request merged**
